std/range/primitives.d

```diff
@@ -1987,6 +1987,81 @@ ElementType!R moveAt(R, I)(R r, I i) if (isIntegral!I)
 }
 
 /**
+Pops the first element from $(D r) and returns it.
+It combines $(D front) and $(D popFront) in one method.
+
+---
+R r = [1,2,3];
+assert(r.frontPop == 1)
+assert(r == [2,3])
+---
+ */
+auto frontPop(Range)(ref Range r)
```

Comment:
`ref` is needed so that by-value ranges will work as expected. Otherwise odd things may happen.